kernel/src/scan/mod.rs

```diff
@@ -218,20 +214,34 @@ impl Scan {
                 let read_result = if partition_fields.is_empty() {
                     read_result
                 } else {
-                    let mut fields = Vec::with_capacity(partition_fields.len() + len);
+                    debug!("Adding back in partition columns");
+                    let final_schema = self.schema();
+                    let mut fields = vec![None; final_schema.fields.len()];
+                    for select_field in read_schema.fields() {
+                        let field_position = final_schema
+                            .field_index(&select_field.name)
+                            .ok_or_else(|| Error::missing_column(&select_field.name))?;
+                        fields[field_position] = Some(Expression::column(select_field.name()));
```

Comment:
qq: When to use `&field.name` vs. `field.name()`?

Reply:
It's odd that there would be an accessor if the field is `pub`. Let me see if should just make it not pub or maybe remove the method